File: uber/uber_ood/LC359LoggerRateLimiter.py
class Logger(object):

    def __init__(self):
        self.dictionary = {}

    def shouldPrintMessage(self, timestamp, message):
        if message not in self.dictionary:
            self.dictionary[message] = timestamp
            return True
        else:
            if timestamp - self.dictionary[message] < 10:
                return False
            else:
                self.dictionary[message] = timestamp
                return True



# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)

File: apple/LC359LoggerRateLimiter.java
class Logger {
    HashMap<String, Integer> hm;
    public Logger() {
        hm = new HashMap<>();
    }

    public boolean shouldPrintMessage(int timestamp, String message) {
        if(!hm.containsKey(message) || timestamp - hm.get(message) >= 10) {
            hm.put(message, timestamp);
            return true;
        }
        return false;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * boolean param_1 = obj.shouldPrintMessage(timestamp,message);
 */
